testing/integration/upgrade_test.go

```diff
@@ -76,6 +76,7 @@ func TestFleetManagedUpgrade(t *testing.T) {
 		Isolate: false,
 		Sudo:    true, // requires Agent installation
 	})
+	t.Skip("Flaky test: https://github.com/elastic/elastic-agent/issues/3154")
```

Comment:
We lose coverage of Fleet upgrades entirely by disabling this, which I don't like. I don't think we should be merging changes that could break this.

We can either leave it as flaky and hope changes get past it eventually, or we are getting close to having to prevent merges while we work through the test flakiness.

Reply:
ok, then let's merge the PRs fixing other tlaky tests that are blocked because of this flaky test?